### DISCUSSION TITLE
Command hooks

### DISCUSSION THREAD

**Discussion by localusercamp** [laravel/framework#48898] [Ideas] [answered]
Hello everyone.

# Introduction
Recently I had to do an optimization in my project. 
I have many Model route bindings which auto discovered through [RecursiveDirectoryIterator](https://www.php.net/manual/en/class.recursivedirectoryiterator.php) (wrote by hand).
What i want to do is cache the discover step like `Routes` and `config` do in framework itself.

I can create a artisan command for this specific task, but then I will have to remember to run it alongside `artisan optimize`.

# Idea
I thinks it would be cool to be able to subscribe to artisan commands through robust laravel Event system.
This opportunity also will give package developers more flexability. For example: some package might add new console `make:` command which may depend on Model creation. To solve this you could subscribe on `artisan make:model` command and call your own command inside event.

**Comment by NathanaelGT** [comment, marked as answer]
Have you tried this?

```php
use Illuminate\Console\Events\CommandFinished;
use Illuminate\Support\Facades\Event;

Event::listen(function (CommandFinished $event) {
    if ($event->command === 'optimize') {
        //
    }
});
```

**Reply by localusercamp** [reply]
Nope!
I'll definitely try this, thanks!
